hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheRecordStore.java

```diff
@@ -1116,6 +1116,7 @@ public void putRecord(Data key, CacheRecord record, boolean updateJournal) {
     }
 
     protected R doPutRecord(Data key, R record, String source, boolean updateJournal) {
+        markExpirable(record.getExpirationTime());
```

Comment:
It may be a good idea to mark a record store as expirable, but not start the background expiration task until the record store becomes primary. Looks like the later may be started in `AbstractCacheRecordStore#init`.

Reply:
Good idea. I added expiration task start at `init` as you pointed out.